### PR TITLE
[FIXED JENKINS-34172] Add null safety checks to FilePath

### PR DESCRIPTION
**Pull request by slgriff**
See [JENKINS-34172](https://issues.jenkins-ci.org/browse/JENKINS-34172).

### CODE REVIEW

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
Not sure it is a right idea for all methods.
AFAICT it some cases it makes the operations noop when `null` is passed. Instead of that I would suggest throwing checked IOException so that the error does not get suppressed.

But maybe a better way is to add `@Nonnull` annotations and track down incorrect usages (at least within the core)

**Comment by daniel-beck** [comment]
Agree with Oleg, this significantly changes the semantics in case of misuse.

**Comment by slgriff** [comment]
Thanks for the feedback. I will make the requested changes and update the PR.

**Comment by slgriff** [comment]
FIXED JENKINS-34172] Update FilePath with static analysis annotations for null safety checks and to propogate runtime NPEs

File: core/src/main/java/hudson/FilePath.java [hunk 1]
```diff
@@ -635,7 +679,8 @@ private void unzip(File dir, InputStream in) throws IOException {
         }
     }
 
-    private void unzip(File dir, File zipFile) throws IOException {
+    private void unzip(@Nonnull File dir, File zipFile) throws IOException {
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/FilePath.java]: second arg should be nonnull as well

File: core/src/main/java/hudson/FilePath.java [hunk 2]
```diff
@@ -279,6 +280,7 @@ public FilePath(@Nonnull File localPath) {
      * @param rel a path which if relative will be resolved against base
      */
     public FilePath(@Nonnull FilePath base, @Nonnull String rel) {
+    	assert base != null;
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/FilePath.java]: something is wrong with indenting here and below

File: core/src/main/java/hudson/FilePath.java [hunk 3]
```diff
@@ -415,7 +448,8 @@ public void zip(OutputStream os) throws IOException, InterruptedException {
         zip(os,(FileFilter)null);
     }
 
-    public void zip(FilePath dst) throws IOException, InterruptedException {
+    public void zip(@Nonnull FilePath dst) throws IOException, InterruptedException {
+        assert dst != null;
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/FilePath.java]: I do not think asserts are really required, static analysis should start producing issues even before the tests run (we also need to configure surefire to do the same if possible, like IDEA does)

**Review by oleg-nenashev — COMMENTED** [review]
will try to review after DevOps World | Jenkins World. Sorry for the delay, and thanks for your patience!

**Comment by oleg-nenashev** [comment]
I believe that all asserts should be removed. Although I fixed the merge conflict, I do not think this PR can be cleaned up easily. Might be better to start from scratch

@slgriff Are you still interested in it?

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
requesting changes as per comments above

**Comment by daniel-beck** [comment]
No response to comments in two months, so closing as inactive.

@slgriff If you want to resume working on this, just open a new PR with conflicts resolved. Thanks!

**Pull request closed without merging**